spaceship/lib/spaceship/portal/certificate.rb

```diff
@@ -241,29 +270,46 @@ def factory(attrs)
           klass.new(attrs)
         end
 
+        # <b>DEPRECATED:</b> Use <tt>all_by_platform</tt> instead.
         # @param mac [Bool] Fetches Mac certificates if true. (Ignored if callsed from a subclass)
         # @return (Array) Returns all certificates of this account.
         #  If this is called from a subclass of Certificate, this will
         #  only include certificates matching the current type.
         def all(mac: false)
+          puts '`all` is deprecated. Please use `all_by_platform` instead.'.red
+          all_by_platform(platform: mac ? Spaceship::Portal::App::MAC : Spaceship::Portal::App::IOS)
+        end
+
+        # @param platform [String] Fetches platform certificates. (Ignored if called from a subclass)
+        # @return (Array) Returns all certificates of this account.
+        #  If this is called from a subclass of Certificate, this will
+        #  only include certificates matching the current type.
+        def all_by_platform(platform: Spaceship::Portal::App::IOS)
```

Comment:
@bassrock Hey, so some work that's been going on internally is making a request builder object that would store what platform you wanted to use for a given call in the different clients we expose. It would remove the need to create by_platform methods for everything. We can either make a branch from your changes to add the builder or create a work in progress branch get your input put on the builder then push that to master. Thoughts on this?